### ISSUE TITLE
Continuous Build Crashing When Opened (Mac)

### ISSUE THREAD

**Issue by Rob59er** [endless-sky/endless-sky#5013]
MacOS 10.14.16

Downloaded the continuous build off of GitHub, upon opening the .app it promptly crashed without even showing the loading screen. Downloaded the game again and met the same errors. Error is as followed: 
> Dyld Error Message:
  Symbol not found: _objc_opt_respondsToSelector
  Referenced from: /Users/USER/Desktop/Endless Sky.app/Contents/MacOS/../Frameworks/libSDL2-2.0.0.dylib (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libobjc.A.dylib

Here is the details of the thread that crashed:
> Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   dyld                          	0x00000001117183ae __abort_with_payload + 10
1   dyld                          	0x0000000111717ba0 abort_with_payload_wrapper_internal + 82
2   dyld                          	0x0000000111717bd2 abort_with_payload + 9
3   dyld                          	0x00000001116d7a9d dyld::halt(char const*) + 343
4   dyld                          	0x00000001116d7bc7 dyld::fastBindLazySymbol(ImageLoader**, unsigned long) + 167
5   libdyld.dylib                 	0x00007fff74c4a32e dyld_stub_binder + 282
6   ???                           	0x0000000107704140 0 + 4419764544
7   libSDL2-2.0.0.dylib           	0x00000001076a23ad SDL_CreateWindow_REAL + 981
8   Endless-Sky                   	0x00000001072e07c7 GameWindow::Init() + 343
9   Endless-Sky                   	0x000000010735a710 main + 1312
10  Endless-Sky                   	0x000000010729c934 start + 52

Reinstalled SDL2 with no success.
As for reproduction I have no advice.
If you need further information or details from the error file feel free to ask.

**Comment by starmlerp** [comment]
..Yeah, the continuous builds are... continuous builds. Even the disclaimer states that it may not work or malfunction in any way. Unless this keeps happening when the official release comes out, I would recommend you to close this issue.

**Comment by tehhowch** [comment]
Without a Mac to work from, this will probably stay open a while.

**Comment by MCOfficer** [comment]
Can you give more information regarding your system? macOS version, stuff  installed via homebrew (if any), you know the drill.

**Comment by MCOfficer** [comment]
I actually did get a similar report from CallMeShmae some time ago:
```
Dyld Error Message:
  Symbol not found: _objc_alloc_init
  Referenced from: /Applications/Endless Sky.app/Contents/MacOS/../Frameworks/libSDL2-2.0.0.dylib
  Expected in: /usr/lib/libobjc.A.dylib
```
That was on macOS 10.12.6 (Sierra).

---

I assume this is caused by our build target, which afaik is implicitly set to Catalina (10.15.4). We can adjust that via the `MACOSX_DEPLOYMENT_TARGET` environment variable ([Example](https://github.com/EndlessSkyCommunity/ESLauncher2/blob/01afa3599afc512eb90464000019f666d8352a0b/.github/workflows/build.yaml#L49-L52)).
However, i doubt this will help, since the issue is apparently here:
> ... libSDL2-2.0.0.dylib (which was built for Mac OS X 10.15)

So as long as we build on Catalina (our only option), i'd assume that we can only bundle libraries built for Catalina. Which would limit the continuous builds to Catalina only.

**Comment by MCOfficer** [comment]
I thought i had updated this issue, but apparently i didn't.

I investigated this a bit further, but didn't manage to make any improvements. It all comes down to this:
- We install dependencies via homebrew, because compiling all of them would take forever
- Homebrew only offers binaries built for the system you have (in this case catalina), with no backwards compatiblity
- Some packages can be built from source, however it seems impossible to pass in environment variables such as `MACOSX_DEPLOYMENT_TARGET`

---

Bottom line:
1. We use libraries that aren't compatible with older OSX versions
2. We have no reasonable way of getting compatible libraries (?)
3. Shit.

**Comment by thebigh2014** [comment]
From my conversation with @MCOfficer and @tehhowch regarding the ESL2 ( https://github.com/EndlessSkyCommunity/ESLauncher2/issues/64 ), I've been working on trying to get the Github CI/CD to use an older deployment target.  Will continue to report progress and discuss further here.

Things tried already:

- setting the ENV variable ```MACOSX_DEPLOYMENT_TARGET``` in the Xcode project to 10.11 (my current OS)
- using Xcode 10.3 on the CI/CD VM (which uses the macOS 10.14 SDK)

Things will be trying:

- setting the ENV variable in the compile/build step

One more thing, the error I'm encountering is the one mentioned above in https://github.com/endless-sky/endless-sky/issues/5013#issuecomment-619352506  This means that my build may work on one or more of the OSes between mine (10.11) and Catalina (10.15).

**Comment by thebigh2014** [comment]
I've realized that since homebrew pulls in binary dependencies for the VM's OS, this is likely why the failure in `libSDL2-2.0.0.dylib` keeps happening.  Along with setting the ENV variable, I will also be adding the option to homebrew to build dependencies from source rather than using pre-compiled binaries as there is no option to pull in binaries for a specific OS version.  There are further options with homebrew to control the entire build process of dependencies (execute commands with custom arguments) but I'm thinking that will be a bit excessive.  I'll post again when I have some results (one way or the other).

**Comment by tehhowch** [comment]
This should be resolved now, due to the work done in the ci build and XCode project dependencies by EricFromCanada.

@thebigh2014 @Rob59er please verify.

**Comment by thebigh2014** [comment]
@tehhowch I can confirm that the continuous build works on Mojave (10.14.6) however since I'm not actively running El Capitan (10.11.x) any longer, I cannot say the same for it.  My advice would be to show/indicate the minimum macOS requirement as Mojave (10.14.6) until someone can confirm it works on earlier systems.

**Comment by dragonmaus** [comment]
Confirmed working on High Sierra (10.13.6):
![Screen Shot 2021-11-05 at 2 11 05 PM](https://user-images.githubusercontent.com/5817605/140559407-2de8a4bd-0389-404b-8cbc-981774bb5a59.png)
![Screen Shot 2021-11-05 at 2 11 54 PM](https://user-images.githubusercontent.com/5817605/140559417-e6ea5709-f4ef-4c9b-95df-b1f12ab113b4.png)
